packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -141,22 +157,32 @@ export class State {
   }
 
   private toStateJson(scope: Construct, actionBindOptions: ActionBindOptions): CfnDetectorModel.StateProperty {
-    const { onEnter } = this.props;
+    const { onEnter, onInput, onExit } = this.props;
     return {
       stateName: this.stateName,
-      onEnter: onEnter && { events: toEventsJson(scope, actionBindOptions, onEnter) },
+      onEnter: {
+        events: toEventsJson(scope, actionBindOptions, onEnter),
+      },
       onInput: {
```

Comment:
Can we only render this `OnInput` object if `onInput` is not `undefined`, and `this.transitionEvents` is not empty?